lib/util/stream_utils.js

```diff
@@ -400,6 +400,22 @@ shaka.util.StreamUtils = class {
     return tracksChanged;
   }
 
+  /**
+   * Cache browser engine call to improve performance on some poor platforms
+   *
+   * @param {string} type
+   * @return {boolean}
+   */
+  static getMediaSourceSupportedType(type) {
```

Comment:
I think, design-wise, this might be better as a static method on `MediaSourceEngine`.
That class exists to be our wrapper for MediaSource operations, so it would make more sense to hold a cache like this.

It also contains a few calls to `MediaSource.isTypeSupported`, which we could cache. Perhaps not the call in `MediaSourceEngine.probeSupport` though, so that we don't use memory uselessly on types that are not actually in the manifest.

Reply:
Yeah, that is a better way.
I did so because `MediaSourceEngine.probeSupport` and `MediaSourceEngine.isStreamSupported` seem depreciated and never called by other places.
Now changed this.